spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -275,18 +275,20 @@ public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchT
 
   @Override
   public boolean dropTable(Identifier ident) {
-    // no need to check table existence to determine which catalog to use. if a table doesn't exist
```

Comment:
This PR requires tests that would show the exact problem that was faced and how the new logic behaves now.

Reply:
> This PR requires tests that would show the exact problem that was faced and how the new logic behaves now.

thanks, added new unit test

Reply:
Based on what I see in the test case, I don't think the problem is in `SparkSessionCatalog`. The primary issue is that our `HiveCatalog` may drop a pointer without checking if it points to a valid Iceberg table.

Reply:
I think it is acceptable that using the iceberg internal catalog to delete non-Iceberg tables does not behave as expected. Since we've explicitly used iceberg catalog this means we can think of it as an iceberg table (although this is unreliable).
But we must ensure that `SparkSessionCatalog` behaves as expected, so adding judgment in `SparkSessionCatalog` also makes sense to me.